app/gui2/src/components/ExecutionModeSelector.vue

```diff
@@ -0,0 +1,105 @@
+<script setup lang="ts">
+import WorkflowPlayIcon from '@/assets/icons/workflow_play.svg'
+
+import { useDocumentEvent } from '@/util/events'
+
+import { ref } from 'vue'
+
+const props = defineProps<{ modes: string[]; modelValue: string }>()
+const emit = defineEmits<{ execute: []; 'update:modelValue': [mode: string] }>()
+
+const isDropdownOpen = ref(false)
+
+const executionModeSelectorNode = ref<HTMLElement>()
+const executionModeDropdownNode = ref<HTMLElement>()
+
+function onDocumentClick(event: MouseEvent) {
+  if (
+    event.target instanceof Node &&
+    executionModeSelectorNode.value?.contains(event.target) === false &&
+    executionModeDropdownNode.value?.contains(event.target) === false
```

Comment:
I don't think two element refs are required here. Just one check on the component root node would suffice.